src/batch-client/index.ts

```diff
@@ -1335,15 +1341,26 @@ export class ZimbraBatchClient {
 			singleRequest: true
 		});
 
-	public modifyPrefs = (prefs: PreferencesInput) =>
-		this.jsonRequest({
+	public modifyPrefs = (prefs: PreferencesInput) => {
+		let attrs: any = mapValuesDeep(prefs, coerceBooleanToString);
+		Object.keys(attrs).map(pref => {
```

Comment:
If there is only the need of iteration of the array, we can use forEach instead of map. 
And we can also skip `Object.keys()` by using for...in loop for the object.

Reply:
Object is object not array

Reply:
updated